[16.0][FW][FIX] mail_show_follower: Use company from the document to render messages correctly + proper lang handle

carlos-lopez-tecnativa opened the pull request:
Forward-port of  https://github.com/OCA/social/pull/1464
In a multi-company environment, if one company uses the show_followers_partner_format by partner_name and another company uses partner_email, a user with permissions in both companies may experience issues. If a document belongs to Company A, but the current company is set to Company B, the message format is taken from the current company (Company B) instead of the document’s company (Company A). This fix ensures that the message format is rendered based on the document's company.

It also fixes the lang handling for cases with False values.

TT51022
@Tecnativa @pedrobaeza @yajo @chienandalu

pedrobaeza commented on mail_show_follower/static/description/index.html:
```diff
@@ -8,11 +8,10 @@
 /*
 :Author: David Goodger (goodger@python.org)
-:Id: $Id: html4css1.css 9511 2024-01-13 09:50:07Z milde $
+:Id: $Id: html4css1.css 8954 2022-01-20 10:10:25Z milde $
```
This shouldn't be changed.

carlos-lopez-tecnativa committed:
[FIX] mail_show_follower: Use company from the document to render messages correctly + proper lang handle

In a multi-company environment, if one company uses the
show_followers_partner_format by partner_name and another company
uses partner_email, a user with permissions in both companies may
experience issues. If a document belongs to Company A, but the current
company is set to Company B, the message format is taken from the
current company (Company B) instead of the document’s company
(Company A). This fix ensures that the message format is rendered based
on the document's company.

It also fixes the lang handling for cases with False values.

yajo reviewed (approved):
Thanks!

/ocabot merge patch
